feel-repl.sc

```diff
@@ -7,18 +7,40 @@ import $ivy.`org.apache.logging.log4j:log4j-slf4j-impl:2.14.0`, org.apache.loggi
 Configurator.setRootLevel(Level.WARN)
 
 // initialize the FEEL engine
-val feelEngine = new FeelEngine() 
+val feelEngine = new FeelEngine()
 
 // define a shortcut function for evaluation
 def feel(expression: String, context: Map[String, Any] = Map.empty) {
   feelEngine.evalExpression(expression, context) match {
     case Right(result) => println(fansi.Color.LightGreen(s"> $result"))
-    case Left(failure) => println(fansi.Color.LightRed(s"> $failure"))  
+    case Left(failure) => println(fansi.Color.LightRed(s"> $failure"))
   }
 }
 
+def feel(expression: String, jsonContext: String) {
+  val json = ujson.read(jsonContext) match {
+    case ujson.Obj(objValue) => {
+      val context = objValue.map { case (key, value) => key -> unpackJson(value)}.toMap
+      feel(expression, context)
+    }
+    case otherValue => println(fansi.Color.LightRed(s"> Expected a JSON object as variables but found '$otherValue'"))
+  }
+}
+
+def unpackJson(json: ujson.Value): Any = {
```

Comment:
I was wondering whether the return type could be improved here. Does scala support union types like `Map | List | primitive`?

Reply:
Not directly. Maybe, using a sealed type. But for this case, I don't see much value.